src/swap/SwapScreenV2.tsx

```diff
@@ -0,0 +1,1149 @@
+
+        if (!bothTokensPresent || amountIsTooSmall) {
+          return
+        }
+
+        // This variable prevents the quote from needlessly being fetched again.
+        const quoteIsTheSameAsTheLastOne =
+          quote &&
+          quote.toTokenId === toToken.tokenId &&
+          quote.fromTokenId === fromToken.tokenId &&
+          processedAmountsFrom.token.bignum &&
+          quote.swapAmount.eq(processedAmountsFrom.token.bignum)
+
+        if (!quoteIsTheSameAsTheLastOne) {
+          replaceAmountTo('')
+          void refreshQuote(
+            fromToken,
+            toToken,
+            { FROM: processedAmountsFrom.token.bignum, TO: null },
+            Field.FROM
+          )
+        }
+      }, FETCH_UPDATED_TRANSACTIONS_DEBOUNCE_TIME_MS)
+
+      return () => {
+        clearTimeout(debounceTimeout)
+      }
+    },
+    [
+      processedAmountsFrom.token.bignum?.toString(),
+      fromToken,
+      toToken,
+      quote,
+      refreshQuote,
+      // clearQuote,
+      // replaceAmountTo,
+    ]
+  )
+
+  useEffect(function trackSwapScreenOpen() {
+    AppAnalytics.track(SwapEvents.swap_screen_open)
+  }, [])
+
+  useEffect(
+    function trackImpactWarningDisplayed() {
+      if (warnings.showPriceImpactWarning || warnings.showMissingPriceImpactWarning) {
+        if (!quote) {
+          return
+        }
+        const fromToken = tokensById[quote.fromTokenId]
+        const toToken = tokensById[quote.toTokenId]
+
+        if (!fromToken || !toToken) {
+          // Should never happen
+          Logger.error(TAG, 'fromToken or toToken not found')
+          return
+        }
+
+        AppAnalytics.track(SwapEvents.swap_price_impact_warning_displayed, {
+          toToken: toToken.address,
+          toTokenId: toToken.tokenId,
+          toTokenNetworkId: toToken.networkId,
+          toTokenIsImported: !!toToken.isManuallyImported,
+          fromToken: fromToken.address,
+          fromTokenId: fromToken.tokenId,
+          fromTokenNetworkId: fromToken?.networkId,
+          fromTokenIsImported: !!fromToken.isManuallyImported,
+          amount: processedAmountsFrom.token.bignum
+            ? processedAmountsFrom.token.bignum.toString()
+            : '',
+          amountType: 'sellAmount',
+          priceImpact: quote.estimatedPriceImpact,
+          provider: quote.provider,
+        })
+      }
+    },
+    [warnings.showPriceImpactWarning || warnings.showMissingPriceImpactWarning]
+  )
+
+  function onOpenTokenPickerFrom() {
+    AppAnalytics.track(SwapEvents.swap_screen_select_token, { fieldType: Field.FROM })
+    // use requestAnimationFrame so that the bottom sheet open animation is done
+    // after the selectingField value is updated, so that the title of the
+    // bottom sheet (which depends on selectingField) does not change on the
+    // screen
+    requestAnimationFrame(() => tokenBottomSheetFromRef.current?.snapToIndex(0))
```

Comment:
1) shall we consistenly use the `hadndle` prefix?
2) perhaps we can extract this code to a separate function e.g. `handleOpenTokenPicker(fieldType: Field)` to emphasize that hanlding is mostly identical